scratchwork/main.go

```diff
@@ -59,27 +95,62 @@ func getDeckList() (StaticDeckList, error) {
 	return sdl, nil
 }
 
+func getCardPos() (PositionalRectangles, error) {
+	cardPos := PositionalRectangles{}
+
+	endpoint := fmt.Sprintf("%s/positional-rectangles", legendHost)
+	resp, err := http.Get(endpoint)
+	if err != nil {
+		return cardPos, errors.Wrap(err, "getting card positions failed")
+	}
+
+	dec := json.NewDecoder(resp.Body)
+	err = dec.Decode(&cardPos)
+	if err != nil {
+		return cardPos, errors.Wrap(err, "response could not be interpreted as card positions")
+	}
+	return cardPos, nil
+}
+
+func convertDeckListCodesToNames(list StaticDeckList) StaticDeckList{
```

Comment:
There should be a way to determine if we need to do this or not. There's a lot of different methods to accomplish this, mainly I would think of hashing values and using those. Otherwise, using a cached boolean is fine.

Reply:
CardsInDeck should have a boolean field telling us if we've translated them

Reply:
```
func (list StaticDeckList) convertCardCodesToName() error {
  ...
}
```